src/DotNetWorker.Grpc/NativeHostIntegration/NativeMethods.cs

```diff
@@ -31,15 +41,39 @@ public static void SendStreamingMessage(NativeSafeHandle nativeApplication, Stre
             _ = send_streaming_message(nativeApplication, bytes, bytes.Length);
         }
 
-        [DllImport(NativeWorkerDll)]
+        [DllImport(NativeWorkerDll, CharSet = CharSet.Auto)]
         private static extern int get_application_properties(out NativeHost hostData);
 
-        [DllImport(NativeWorkerDll)]
+        [DllImport(NativeWorkerDll, CharSet = CharSet.Auto)]
         private static extern int send_streaming_message(NativeSafeHandle pInProcessApplication, byte[] streamingMessage, int streamingMessageSize);
 
-        [DllImport(NativeWorkerDll)]
+        [DllImport(NativeWorkerDll, CharSet = CharSet.Auto)]
         private static extern unsafe int register_callbacks(NativeSafeHandle pInProcessApplication,
             delegate* unmanaged<byte**, int, IntPtr, IntPtr> requestCallback,
             IntPtr grpcHandler);
+
+        /// <summary>
+        /// Custom import resolve callback.
+        /// When trying to resolve "FunctionsNetHost", we return the handle using GetMainProgramHandle API in this callback.
+        /// </summary>
+        private static IntPtr ImportResolver(string libraryName, System.Reflection.Assembly assembly, DllImportSearchPath? searchPath)
+        {
+            if (libraryName == NativeWorkerDll)
+            {
+#if NET6_0
+                if (OperatingSystem.IsLinux())
+                {
+                    return NativeLibraryLinux.GetMainProgramHandle();
+                }
+#elif NET7_0_OR_GREATER
+                return NativeLibrary.GetMainProgramHandle();
+#else
+                throw new PlatformNotSupportedException("Interop communication with FunctionsNetHost is not supported in the current platform. Consider upgrading your project to .NET 7.0 or later.");
```

Comment:
Doesn't .NET 6 and linux also work? Or are we trying to push for users to use net7 and greater with this error message?

Reply:
For .NET 6 & Linux : `NativeLibraryLinux.GetMainProgramHandle()` return value will be returned.
For .NET 6 & Non Linux (Windows): Will return `return IntPtr.Zero;` line 85. So  built-in resolving code will be used (which works for Windows).
For .NET 7or greater (Both Win & Linux OS): `NativeLibrary.GetMainProgramHandle()` method (which is cross platform) and available from net7 onwards, will be used.

Reply:
Would be good to add a comment to some of this logic. It's fresh in our heads right now, but having the details documented (including the fact the new API is only available on .NET 7+) will help in the future.

Reply:
I have documented that information in the Shim class (NativeLibraryLinux.cs).

Reply:
Updated here as well now.